test/e2e/delete_test.go

```diff
@@ -51,9 +36,8 @@ func testWindowsNodeDeletion(t *testing.T) {
 	require.NoError(t, err, "Windows node deletion failed")
 
 	// Cleanup all the MachineSets created by us.
-	for _, machineSet := range e2eMachineSets {
-		assert.NoError(t, testCtx.deleteMachineSet(machineSet), "error deleting MachineSet")
-	}
+	assert.NoError(t, testCtx.ensureMachineSetIsDeleted(machineSetWithLabel), "error deleting MachineSet")
+	assert.NoError(t, testCtx.ensureMachineSetIsDeleted(machineSetWithoutLabel), "error deleting MachineSet")
```

Comment:
can use a different error message here which will help to differentiate if the tests fail.